File: my-folder/1777-determine-if-two-strings-are-close/solution.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if (word1.length() != word2.length()) {
            return false;
        }
        int[] freq1 = new int[26];
        int[] freq2 = new int[26];
        for (int i = 0; i < word1.length(); i++) {
            freq1[word1.charAt(i) - 'a']++;
            freq2[word2.charAt(i) - 'a']++;
        }
        for (int i = 0; i< 26; i++) {
            if ((freq1[i] == 0 && freq2[i] !=0) || (freq1[i] != 0 && freq2[i] ==0)) {
                return false;
            }
        }
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < 26; i++) {
            int count1 = map.getOrDefault(freq1[i], 0);
            map.put(freq1[i], count1+1);
            int count2 = map.getOrDefault(freq2[i], 0);
            map.put(freq2[i], count2-1);
        }
        for (int i : map.values()) {
            if (i != 0) {
                return false;
            }
        }
        return true;
    }
}

File: my-folder/1777-determine-if-two-strings-are-close/solution.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        freq1 = [0] * 26
        freq2 = [0] * 26
        for c in word1:
            freq1[ord(c) - ord('a')] += 1
        for c in word2:
            freq2[ord(c) - ord('a')] += 1
        hash_map = {}
        for i in range (0, 26):
            w1, w2 = freq1[i], freq2[i]
            if w1 == 0 and w1 != 0 or w1 != 0 and w2 == 0:
                return False
        for freq in freq1:
            if freq not in hash_map:
                hash_map[freq] = 0
            hash_map[freq] += 1
        for freq in freq2:
            if freq not in hash_map:
                hash_map[freq] = 0
            hash_map[freq] -= 1
        for value in hash_map.values():
            if value != 0:
                return False
        return True
